lib/resource-serializer.js

```diff
@@ -1,9 +1,22 @@
 'use strict';
+var _ = require('lodash');
 var SerializerUtils = require('./serializer-utils');
 
 function ResourceSerializer(collectionName, record, opts) {
   var payload = { data: {} };
 
+  function getLinks(links) {
+    return _.mapValues(links, function (value) {
+      if (_.isFunction(value)) {
+        return value(records);
+      } else {
+        return value;
+      }
+    });
+  }
+
+  if (opts.topLevelLinks) { payload.links = getLinks(opts.topLevelLinks); }
+
```

Comment:
I looked at that, but since `collection-serializer.js` performs the iteration over the objects, `serializer-utils` is not able to add properties to the top level. I guess this requires some more refactoring to get it done. Mixing up instance with static functions in `serializer-utils.js` doesn't sound like a very clean solution.